tests/unittest.py

```diff
@@ -187,6 +245,39 @@ def assertConsensusEqual(self, manager1, manager2):
             # Hard verification
             # self.assertEqual(tx1_meta.voided_by, tx2_meta.voided_by)
 
+    def assertConsensusEqualSyncV2(self, manager1, manager2):
+        # The current sync algorithm does not propagate voided blocks/txs
+        # so the count might be different even though the consensus is equal
+        # One peer might have voided txs that the other does not have
```

Comment:
I didn't understand the current implementation. I feel we should assert that the best block is the same and that the mempool is the same.

Reply:
I'll do that.

Reply:
I just checked, that is already being done. But there are additional checks, in particular, when comparing the consensus of nodes A and B:

- any voided transaction that node A has must either not exist in node B or must also be voided in node B; (and vice-versa)
- any non-voided transaction that node A has must either not exist in node B or must not be voided in node B; (and vice-versa)

But we already do check that the best block is the same and that the mempool-tips (and hence the mempool) is the same. So the second condition might not make too much sense, since all non-voided transactions must exist in both, I could actually change it to check that (actually I just quick tested doing it and it seems OK).

Reply:
I refactored and simplified it a little bit. I couldn't run a large batch of tests locally, so I'll wait for the tests to run on the CI.

Reply:
Outdated.